File: StoreCredit/StoreCredit.js
var lines;
var output = "";

fs = require("fs");
fs.readFile("C-large-practice.in", "utf-8", function(err, data){
	if (err){
		return console.log(err);
	}
	lines = data.split("\n");
	var D = +lines[0];
	for(var i = 0; i < D; i++){
		output += "Case #" + (i+1) + ": "
		var credit = +lines[(3*i)+1];
		var items = +lines[(3*i)+2];
		var pricesr = lines[(3*i)+3].split(" ");
		var prices = [];
		for(var a = 0; a < items; a++){
			prices[a]= +pricesr[a];
		}
		for(var j = 0; j < items; j++){
			for(var k  = j+1; k < items; k++){
				if(prices[j]+prices[k] === credit){
					output += (j+1) + " " + (k+1);
					break;
				}

			}
		}
		output += "\n";
	}		
	fs.writeFile("OutputC.txt", output, function(err){
		if(err){
			return console.log(err);
		}
		console.log("File was saved!");
	});
});

File: StoreCredit/StoreCredit.exs
textr = File.read! "C-small-practice.in"
text = String.split(textr, "\n")
cases = String.to_integer(Enum.fetch!(text,0))

File.rm("StoreCreditEX.txt")

credit = Enum.concat([0,0],text) |> Enum.take_every(3) |> Enum.drop(1)
items = Enum.concat([0],text) |> Enum.take_every(3) |> Enum.drop(1) |> Enum.map(fn x -> String.to_integer(x) end)
pricesr = Enum.take_every(text,3) |> Enum.drop(1)

prices = Enum.map(pricesr, fn x -> String.split(x," ") |> Enum.map(fn x -> String.to_integer(x) end)end)
answer = []

for i <- 0..cases-1 do
	for j <- 0..Enum.fetch!(items, i)-2 do
		for k <- j+1..Enum.fetch!(items, i)-2 do
			price1 = Enum.fetch!(prices, i) |> Enum.fetch!(j)
			price2 = Enum.fetch!(prices, i) |> Enum.fetch!(k)
			if (price1+price2) === (String.to_integer(Enum.fetch!(credit,i))) do
				answer = Enum.concat(answer,["Case ##{i+1}: #{j+1} #{k+1}\n"])
				File.write("StoreCreditEX.txt", answer,[:append])
			end

		end
	end
end